phantomjs-env/src/main/scala/org/scalajs/jsenv/phantomjs/PhantomJSEnv.scala

```diff
@@ -119,37 +92,50 @@ class PhantomJSEnv(config: PhantomJSEnv.Config)
       val ctors = clazz.getConstructors()
       assert(ctors.length == 1, "JettyWebsocketManager may only have one ctor")
 
-      val mgr = ctors.head.newInstance(websocketListener)
+      val listener = new WebsocketListener {
+        def onRunning(): Unit = startPhantomJSProcess()
+        def onOpen(): Unit = sendPendingMessages()
+        def onClose(): Unit = ()
+        def onMessage(msg: String): Unit = receiveFrag(msg)
+        def log(msg: String): Unit = logger.debug(s"PhantomJS WS Jetty: $msg")
+      }
+
+      val mgr = ctors.head.newInstance(listener)
 
       mgr.asInstanceOf[WebsocketManager]
     }
 
     private val mgr: WebsocketManager = loadMgr()
 
-    future.onComplete(_ => synchronized(notifyAll()))(ExecutionContext.global)
-
-    private[this] val recvBuf = mutable.Queue.empty[String]
     private[this] val fragmentsBuf = new StringBuilder
 
-    private def comSetup = {
+    // Constructor
+    mgr.start()
+
+    private def startPhantomJSProcess(): Unit = synchronized {
+      import ExecutionContext.Implicits.global
+
+      val comSetup = makeComSetupFile()
+      underlyingRun = internalStart(comSetup :: files, runConfig)
+      underlyingRun.future.onComplete { result =>
+        promise.tryComplete(result)
+      }
```

Comment:
This probably needs to be synced with `mgr` (so that if the manager is not closed yet, it doesn't complete).